### PR TITLE
Average goals methods da

### PR DESCRIPTION
**Pull request by d-atkins**
- Add average_goals_per_game - returns the average goals across all games
- Add average_goals_by_season - returns a hash with the average goals per season
- Add games_by_season - returns a hash with the game count by season
- Merge all pushed branches

### CODE REVIEW

File: test/game_collection_test.rb [hunk 1]
```diff
@@ -23,6 +23,28 @@ def test_it_can_create_games_from_csv
     assert_equal 3, @game.home_goals
   end
 
+  def test_it_can_calculate_average_goals_per_game
+    expected = 4.31
+    assert_equal  expected, @game_collection.average_goals_per_game
```
Comment (mronauli) [inline test/game_collection_test.rb]: why not just assert equal 4.31 instead of expected?

File: test/game_collection_test.rb [hunk 2]
```diff
@@ -23,6 +23,28 @@ def test_it_can_create_games_from_csv
     assert_equal 3, @game.home_goals
   end
 
+  def test_it_can_calculate_average_goals_per_game
+    expected = 4.31
```
Comment (tylertomlinson) [inline test/game_collection_test.rb]: Looks good. 4.31 just needs to be moved to next line

Comment (d-atkins) [reply test/game_collection_test.rb]: on it

File: test/game_collection_test.rb [hunk 3]
```diff
@@ -23,6 +23,28 @@ def test_it_can_create_games_from_csv
...
+    expected = {"20162017" => 4, "20142015" => 6, "20152016" => 10, "20132014" => 6}
+    assert_equal expected, @game_collection.games_by_season
+  end
+
+  def test_it_can_calculate_average_goals_by_season
+    avg_goals_by_season = @game_collection.average_goals_by_season
+    assert_equal 4.33, avg_goals_by_season["20132014"]
+    assert_equal 3.5, avg_goals_by_season["20142015"]
+    assert_equal 4.6, avg_goals_by_season["20152016"]
+    assert_equal 4.75, avg_goals_by_season["20162017"]
```
Comment (mronauli) [inline test/game_collection_test.rb:44]: looks robust

File: test/game_collection_test.rb (hunk 1 above)
Comment (d-atkins) [reply test/game_collection_test.rb]: good call

**Pull request merged**